vscode-test-runner/workspace-1/.joyride/src/integration_test/index.cljs

```diff
@@ -0,0 +1,52 @@
+(ns integration-test.index
+  (:require [cljs.test]
+            [promesa.core :as p]))
+
+(def !results (atom {:fail 0
+                     :error 0}))
+
+(defmethod cljs.test/report [:cljs.test/default :begin-test-var] [m]
+  (println "===" (-> m :var meta :name))
+  (println))
+
+(def old-fail (get-method cljs.test/report [:cljs.test/default :fail]))
+
+(defmethod cljs.test/report [:cljs.test/default :fail] [m]
+  (old-fail m)
+  (swap! !results update :fail inc))
+
+(def old-error (get-method cljs.test/report [:cljs.test/default :fail]))
+
+(defmethod cljs.test/report [:cljs.test/default :error] [m]
+  (old-error m)
+  (swap! !results update :error inc))
+
+(defn run-all-tests []
+  (p/create (fn [resolve, reject]
+              (p/do!
+               (p/delay 1000) ; This waits for workspace_activate.cljs to run
+               (require '[integration-test.workspace-activate-test])
+
+               ;; This fails when accessing integration-test.run-a-ws-script-test/symbol-1
+               ;; if required from the ns form
+               (require '[integration-test.ws-scripts-test])
+               #_(require '[integration-test.npm-test])
+
+               (cljs.test/run-tests 'integration-test.workspace-activate-test)
+               (cljs.test/run-tests 'integration-test.ws-scripts-test)
+               #_(cljs.test/run-tests 'integration-test.npm-test)
+               
+               ;; Running only matching namespaces fails
+               ;; Error: Doesn't support name: user-activate 
+               ;; (cljs.test/run-all-tests #"integration-test")
+               
+               (println "!results" @!results)
+               (let [{:keys [fail error]} @!results]
```

Comment:
Since `cljs.test/run-test` doesn't return a promise, we're exiting and resolving this promise probably too early. Instead, I think the resolving must happen in the cljs.test multimethods. We can use `promesa.core/deferred` to create a promise which can be resolved inside those multimethods.

Reply:
Not sure I understand how to do this. Want to push the change? Or give me some more clues. 😄

Reply:
Now I think I understood what you meant. Pushed a change.